package-support-policy.md

```diff
@@ -0,0 +1,64 @@
+| System.Reflection.DispatchProxy         | Yes              | <ul><li>.NET Framework 4.6.2+: Provides an implementation of [`DispatchProxy`](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.dispatchproxy).</li><li>.NET 6+: The type is part of the shared framework.</li></ul> | <ul><li>.NET Framework 4.6.2+: Update.</li><li>.NET 6+: Remove.</li></ul> |
+| System.Runtime.CompilerServices.Unsafe  | Yes              | <ul><li>.NET Framework 4.6.2+: Provides the implementation of [`Unsafe`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe).</li><li>.NET 6+: The type is part of the shared framework</li></ul> | <ul><li>.NET Framework 4.6.2+: Update.</li><li>.NET 6+: Remove.</li></ul> |
+| System.Runtime.WindowsRuntime           | No               | Built-in support for WinRT [has been removed from .NET](https://learn.microsoft.com/en-us/dotnet/core/compatibility/interop/5.0/built-in-support-for-winrt-removed).<br />Follow [the recommended actions](https://learn.microsoft.com/en-us/dotnet/core/compatibility/interop/5.0/built-in-support-for-winrt-removed#recommended-action). | Remove. |
+| System.Runtime.WindowsRuntime.UI.Xaml   | No               | Built-in support for WinRT [has been removed from .NET](https://learn.microsoft.com/en-us/dotnet/core/compatibility/interop/5.0/built-in-support-for-winrt-removed).<br />Follow [the recommended actions](https://learn.microsoft.com/en-us/dotnet/core/compatibility/interop/5.0/built-in-support-for-winrt-removed#recommended-action). | Remove. |
+| System.Threading.Tasks.Extensions       | Yes              | <ul><li>.NET Framework 4.6.2+: Provides implementations of [`ValueTask`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask) and [`ValueTask<TResult>`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1).</li><li>.NET 6+: The types are already part of the shared framework.</li></ul> | <ul><li>.NET Framework 4.6.2+: Update.</li><li>.NET 6+: Remove.</li></ul> |
+| System.ValueTuple                       | No               | The .NET Framework implementation contained in the package is simply an in-box façade that forwards the types to the system-installed framework. | Remove. |
```

Comment:
Are you sure about this one? I thought System.ValueTuple is only inbox on .NET 4.7+

Reply:
If I compile this code:
```cs
Console.WriteLine(typeof(System.ValueTuple).Assembly.CodeBase);
```

with this project:
```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFrameworks>net462;net47;net472</TargetFrameworks>
        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
        <OutputType>Exe</OutputType>
    </PropertyGroup>
    <ItemGroup>
      <PackageReference Include="System.ValueTuple" Version="4.5.0" />
    </ItemGroup>
</Project>
```

And then run it against each individual framework (`dotnet run -f <framework>`), I get the same answer for all of them:
```
file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dl
```

In other words, all of them are facades. So the sentence is correct as it is right now.

Reply:
But when I look into the latest package, the `net461` lib assembly has implementation code. Why isn't it used?

Reply:
I don't know, I didn't check. Do we care? net462 is the minimum supported.

Reply:
The System.ValueTuple package has a `net461` and `net47` asset. Even though .NET Framework 4.6.1 is out-of-support, the package still gets used when you run on .NET Framework 4.6.2. Because net461 is the best matching TFM in the package. So yes, we care.

> And then run it against each individual framework (dotnet run -f <framework>), I get the same answer for all 3:
file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dl

That isn't the right workflow to validate whether package assets are still used. All three .NET Framework TFMs that you target in your app run against the same .NET Framework runtime, presumably 4.8.1. You target different versions but at run time they all use the same underlying run time version. When you run your code against the .NET Framework 4.6.2 run time you will see that the System.ValueTuple OOB package is indeed used.

Reply:
> That isn't the right workflow to validate whether package assets are still used

That's the method I've been using for all the assemblies we've migrated. `Type.Assembly.CodeBase` tells me during runtime what DLL is being used.

> You target different versions but at run time they all use the same underlying run time version. When you run your code against the .NET Framework 4.6.2

How can I enforce this? @ericstj what are your thoughts on this? I want to make sure I haven't missed any key validations to ensure the migrated assemblies are targeting the right things.

Reply:
You can test this by adding a console log that prints `System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription` (the API unfortunately requires .NET Framework 4.7.1 or above) you will see that the underlying runtime is not 4.6.2.

Reply:
Looking at https://learn.microsoft.com/en-us/dotnet/framework/migration-guide/versions-and-dependencies#net-framework-462 it seems that 4.6.2 was the default in Server 2016, which I suspect is why it's still supported.  I think if you had a VM with this or a windows container you could see that behavior.

Reply:
Since we have an open issue asking about ValueTuple, we can continue the conversation there: https://github.com/dotnet/maintenance-packages/issues/183